.github/workflows/ci.yml

```diff
@@ -14,357 +17,23 @@ concurrency:
-
-      - name: Install rust toolchain
-        working-directory: ${{ env.RUST_WORKSPACE }}
-        run: rustup show
-
-      - uses: Swatinem/rust-cache@d12701459954fec471b2d34cdf7ea3374b026383 # v1
-        with:
-          working-directory: ${{ env.RUST_WORKSPACE }}
-
-      - name: cargo clippy
-        working-directory: ${{ env.RUST_WORKSPACE }}
-        run: |
-          # workaround for some code being created after rust wants to parse it
-          cargo check --quiet 2>/dev/null || :
-          cargo clippy --all-targets -- --deny warnings
-
-  cargo-test:
-    needs: selection
-    if: ${{ needs.selection.outputs.rust == 'true' }}
-    runs-on: ubuntu-20.04
-    timeout-minutes: 20
-    steps:
-      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
-
-      - name: Install rust toolchain
-        working-directory: ${{ env.RUST_WORKSPACE }}
-        run: rustup show
-
-      - uses: Swatinem/rust-cache@d12701459954fec471b2d34cdf7ea3374b026383 # v1
-        with:
-          working-directory: ${{ env.RUST_WORKSPACE }}
-
-      - name: Run tests
-        working-directory: ${{ env.RUST_WORKSPACE }}
-        run: |
-          # workaround for some code being created after rust wants to parse it
-          cargo check --quiet 2>/dev/null || :
-          cargo test --all-targets --quiet
-          cargo test --doc --quiet
-
-  cargo-doc:
-    needs: selection
-    if: ${{ needs.selection.outputs.rust == 'true' }}
-    runs-on: ubuntu-20.04
-    timeout-minutes: 20
-    steps:
-      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
-
-      - name: Install rust toolchain
-        working-directory: ${{ env.RUST_WORKSPACE }}
-        run: rustup show
-
-      - uses: Swatinem/rust-cache@d12701459954fec471b2d34cdf7ea3374b026383 # v1
-
-      - name: cargo doc
-        working-directory: ${{ env.RUST_WORKSPACE }}
-        run: |
-          # workaround for some code being created after rust wants to parse it
-          cargo check --quiet 2>/dev/null || :
-          cargo doc --all-features --no-deps --document-private-items
-
-  cargo-ok:
-    name: cargo-ok
-    needs:
-      - cargo-format
-      - cargo-clippy
-      - cargo-test
-      - cargo-doc
-    runs-on: ubuntu-20.04
-    steps:
-      - run: exit 0
-
+  dev-ci:
+    uses: xaynetwork/xayn_discovery_engine/.github/workflows/ci_reusable_wf.yml@main
+  
   ci-ok:
     name: ci-ok
-    needs: [selection, dart-ok, cargo-ok]
-    if: ${{always()}}
+    needs: [dev-ci]
     runs-on: ubuntu-20.04
     timeout-minutes: 5
     steps:
       - name: CI OK Testing
         run: |
-          if (${{ needs.selection.outputs.dart == 'true'}} || ${{ needs.selection.outputs.rust == 'true' }} )
+          if (( (${{needs.dev-ci.outputs.rust_output != 'true'}} || ${{needs.dev-ci.outputs.cargo-ok == 'success'}}) && (${{needs.dev-ci.outputs.dart_output != 'true'}} || ${{needs.dev-ci.outputs.dart-ok == 'success'}}) ) || 
```

Comment:
Can't we move these checks once in `ci_reusable_wf` as `ci-ok` and here just refer to that job?

Reply:
Yes, we can.